libshout-idjc: fix missing function declaration

nekopsykose opened the pull request:
usage of str[n]casecmp requires including <strings.h> 
not doing so leads to a implicit-function-declaration warning, which becomes a hard error with gcc14 and clang16

daschuer commented:
Thank you very much. 

I think this should also go to our stable 2.3 branch. 
We merge 2.3 later to main 
Please rebase this accordingly:
```
git rebase --onto=upsteram/2.3 HEAD~1
git push -f
```

In order to distribute your patch with Mixxx, please sign our contributor agreement and comment here when done. 
https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ

nekopsykose commented:
looks like windows needs some other adjustment, no idea what it would be though

daschuer commented:
Ah I see:
https://github.com/mixxxdj/mixxx/blob/f2d670edeb483efb066bf7f00255d01df2f0017c/lib/libshout-idjc/include/os.h#L2
 str[n]casecmp is redefined in Windows. 
So you just need to add a
`#ifndef _MSC_VER` 
Guard

nekopsykose committed:
libshout-idjc: fix missing function declaration

usage of str[n]casecmp requires including <strings.h>
not doing so leads to a implicit-function-declaration warning, which
becomes a hard error with gcc14 and clang16

nekopsykose commented:
aha, right. thanks :)

daschuer reviewed (approved):
The pre-commit issue is unrelated. 
LGTM 
Thank you very much.
